infrastructures/kubernetes/src/main/java/org/eclipse/che/workspace/infrastructure/kubernetes/StartSynchronizer.java

```diff
@@ -137,8 +137,7 @@ public synchronized void start() {
       startTimeMillis = System.currentTimeMillis();
       completionLatch = new CountDownLatch(1);
       eventService.subscribe(runtimeStartInterrupter, KubernetesRuntimeStoppingEvent.class);
-      eventService.subscribe(
-          event -> completionLatch.countDown(), KubernetesRuntimeStoppedEvent.class);
+      eventService.subscribe(runtimeStopWatcher, KubernetesRuntimeStoppedEvent.class);
```

Comment:
Thanks for fixing it.
seems there is no test case for unsubscribing, please consider adding it.